src/Common/src/TypeSystem/IL/Stubs/DelegateMarshallingMethodThunk.Mangling.cs

```diff
@@ -23,13 +24,17 @@ string IPrefixMangledType.Prefix
         {
             get
             {
-                if (IsOpenStaticDelegate)
+                switch (Kind)
```

Comment:
Would it make sense to factor this so that the prefix computation can be shared with the logic in the `Name` property?